echopop/nwfsc_feat/get_proportions.py

```diff
@@ -1,162 +1,1529 @@
+            + ts_length_regression_parameters["intercept"]
+        )
+        / 10.0
+    )
+
+    # Create pivot table for total population (all groups)
+    aggregate_table = utils.create_pivot_table(
+        number_proportions,
+        index_cols=["length_bin"],
+        strat_cols=stratify_by,
+        value_col="proportion",
+    )
+
+    # Calculate total weighted sigma_bs for all strata
+    aggregate_weighted_sigma_bs = (aggregate_table.T * sigma_bs_equiv).T.sum(axis=0)
+
+    # Create pivot table including filter dimensions
+    filtered_population_table = utils.create_pivot_table(
+        number_proportions,
+        index_cols=list(include_filter.keys()) + ["length_bin"],
+        strat_cols=stratify_by,
+        value_col="proportion",
+    )
 
-        # TODO: sketch of ds_proportions
-        - dimensions: stratum, sex, length_bin, age_bin
-        - variables:
-          - abundance_unaged: (stratum, sex, length_bin)
-          - abundance_aaged: (stratum, sex, length_bin, age_bin)
-          - biomass_aged: (stratum, sex, length_bin, age_bin)
-          - biomass_unaged: (stratum, sex, length_bin)
+    # Apply filter to extract target group
+    target_group_table = utils.apply_filters(
+        filtered_population_table, include_filter=include_filter
+    )
 
+    # Aggregate target group over length and strata dimensions
+    target_group_aggregated = target_group_table.unstack("length_bin").sum().unstack(stratify_by)
 
+    # Calculate weighted sigma_bs for target group
+    target_weighted_sigma_bs = (target_group_aggregated.T * sigma_bs_equiv).T.sum(axis=0)
+
+    # Calculate and return proportions
+    return target_weighted_sigma_bs / aggregate_weighted_sigma_bs
+
+
+def get_number_proportions_slice(
+    number_proportions: pd.DataFrame,
+    stratify_by: List[str],
+    exclude_filter: Dict[str, Any] = {},
+    include_filter: Dict[str, Any] = {},
+) -> Union[pd.Series, pd.DataFrame]:
     """
-    ds_proportions: xr.Dataset
-    return ds_proportions
+    Extract number proportions for a specific population slice with flexible filtering.
+
+    This function creates pivot tables from number proportion data and applies
+    inclusion/exclusion filters to extract proportions for specific population
+    segments. Handles multiple stratification dimensions and returns appropriately
+    aggregated results.
+
+    Parameters
+    ----------
+    number_proportions : pd.DataFrame
+        DataFrame with number proportions containing:
+        - proportion: number proportions
+        - length_bin: length bins
+        - Additional grouping columns (age_bin, sex, etc.)
+    stratify_by : List[str]
+        Columns for stratification
+    exclude_filter : Dict[str, Any], default {}
+        Groups to exclude, e.g., {"length_bin": [small_lengths]}
+    include_filter : Dict[str, Any], default {}
+        Groups to include, e.g., {"age_bin": [1], "sex": ["female"]}
```

Comment:
For the 3 `get_*_proportions_slice` functions, only 1 has both `include_filter` and `exclude_filter`, and the other 2 have only `include_filter`. I think that's confusing - better to have them all support both, so user expectation is consistent.